### DISCUSSION TITLE
Wrong Markdown when creating a non-existing page

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#995] [Report a bug] [answered]
If you call a non-existent page in the browser and you are logged in, the editor allows you to create this as a new page. However, the error handler only considers the default template (`page-new-default.md`). For example, if you call a non-existent wiki page and then try to create it, (`page-new-wiki.md`) will not be taken into account and the wiki page will be created as a normal page, even though you are already in the wiki directory. If you go to an already existing wiki page or a normal page created in the wiki directory, creating a new wiki page from there will work fine with the correct template. Tested in a fresh Wiki installation of Datenstrom Yellow. Why does the Error handler behave differently in this case? 

I suspect that it might be due to a technical limitation of the layout system, so I ask this as a question, but maybe this usability glitch can be solved anyway.

**Comment by annaesvensson** [comment]
This is not the intended behaviour. I changed the categorisation to a bug, probably in the [edit extension](https://github.com/annaesvensson/yellow-edit). 

Thank you for finding such an interesting bug. I can confirm the bug with a brand new installation. Select `Small wiki` during the installation, edit the wiki start page in a web browser, make a link to a non-existing wiki page, go to the non-existing wiki page, click on `You can create this page`.  What you see in the web editor is the wrong Markdown file. Here are the version numbers of some of the installed extensions:

```
Datenstrom Yellow 0.9
Core 0.9.12
Edit 0.9.10
Wiki 0.9.4
```

**Comment by annaesvensson** [comment, marked as answer]
A bugfix is now available, I think this fixes the problem reported by Steffen, but there is a possibility we will come back to this in the future. Here's some technical background for software developers. The so-called "path ancestry" is a collection of all parent pages, starting from the home page down to the current page. The path ancestry for a non-existing page was empty, which in itself may not be a surprise, but when combined with various extensions, the results were not always as intended. For example the edit extension could not look through the parent pages and therefore could not find the correct Markdown for a non-existing page. There are at least two ways to fix this problem, in the edit extension `function getRawDataNew()` or in the API `function path()`, we decided to improve the API. With the latest [core extension](https://github.com/annaesvensson/yellow-core) the online editor should work as intended again.

Should be fixed now, please try it out.

**Reply by schulle4u** [reply]
Thank you, works perfectly now on my website. 👍
